app/models/service_plan.rb

```diff
@@ -10,6 +10,10 @@ class ServicePlan < Plan
 
   before_destroy :destroy_contracts
 
+  def self.provided_by(account)
+    Plan.by_type(ServicePlan).issued_by(Service, account.service_ids)
+  end
```

Comment:
This looks like a scope. Maybe define it as a such instead of a class method? Something like:
```suggestion
  scope :provided_by, -> (account) do
    Plan.by_type(ServicePlan).issued_by(Service, account.service_ids)
  end
```

Also, I think that `Plan.by_type(ServicePlan)` returns a list of `ServicePlan` that has access to the `issued_by` scope, so we might be able to use `self` directly. Something like:
```suggestion
  scope :provided_by, -> (account) do
    issued_by(Service, account.service_ids)
  end
```